x/evm/types/tracer.go

```diff
@@ -47,19 +49,37 @@ func GetTracerResult(tracer vm.Tracer, result *core.ExecutionResult) ([]byte, er
 			ReturnValue: returnVal,
 			StructLogs:  FormatLogs(tracer.StructLogs()),
 		})
-	case *tracers.Tracer:
-		res, err = tracer.GetResult()
 	default:
-		res = []byte(fmt.Sprintf("bad tracer type %T", tracer))
+		res, err = tracer.GetResult()
 	}
 	return res, err
 }
 
 // NoOpTracer is an empty implementation of vm.Tracer interface
 type NoOpTracer struct{}
 
+func (dt NoOpTracer) CaptureTxStart(gasLimit uint64) {
+}
+
+func (dt NoOpTracer) CaptureTxEnd(restGas uint64) {
+}
+
+func (dt NoOpTracer) CaptureState(pc uint64, op vm.OpCode, gas, cost uint64, scope *vm.ScopeContext, rData []byte, depth int, err error) {
+}
+
+func (dt NoOpTracer) CaptureFault(pc uint64, op vm.OpCode, gas, cost uint64, scope *vm.ScopeContext, depth int, err error) {
+}
+
+func (dt NoOpTracer) GetResult() (json2.RawMessage, error) {
+	return json2.RawMessage(`{}`), nil
+}
+
+func (dt NoOpTracer) Stop(err error) {
+
+}
+
 // NewNoOpTracer creates a no-op vm.Tracer
-func NewNoOpTracer() *NoOpTracer {
+func NewNoOpTracer() tracers.Tracer {
```

Comment:
no need for nooptracer becasue https://github.com/ethereum/go-ethereum/pull/23708/files add nooptracer already